### ISSUE TITLE
Revise `test_get_job_includes` and endpoint

### ISSUE THREAD

**Issue by diego-plan9** [Qiskit/qiskit-ibmq-provider#133]
### What is the expected behavior?

The `test_get_job_includes` has been [failing](https://travis-ci.com/Qiskit/qiskit-ibmq-provider/jobs/204168475#L797) in a number of recent travis runs for the premium accounts:

```
  File "/home/travis/build/Qiskit/qiskit-ibmq-provider/qiskit/providers/ibmq/api/ibmqconnector.py", line 194, in get_job
    if ('result' in qasm) and ('data' in qasm['result']):
TypeError: argument of type 'NoneType' is not iterable
```

The endpoint has been slightly flaky for a while - we should investigate and coordinate with the API team for a proper solution, and ensure that:

* [ ] the endpoint is working for all environments + both old api and new api
* [ ] the test use the same and correct fields (for connector we use `deleted`, for client we use `shots`)
* [ ] the test also exercises `include_fields`
* [ ] the function handles unexpected cases gracefully (excluding/including non-existent fields, etc)

**Comment by jyu00** [comment]
`get_job` with a filter now fails with 500 internal error. It still works if no filter is specified. The API team opened an issue on this.

**Comment by jyu00** [comment]
`get_job` with `include_fields=['status']` is now failing with a generic error for v1 api. It appears to be the only field failing, and `exclude_fields['status']` works.

**Comment by diego-plan9** [comment]
For tracking purposes, it seems we still [need some tweaks](https://travis-ci.com/Qiskit/qiskit-ibmq-provider/jobs/207482143) for the old API after #154.

**Comment by jyu00** [comment]
v2 api is now working. v1 api will not be fixed as they're being deprecated.

**Comment by diego-plan9** [comment]
Closed by #173 - `master` became green again! :tada:

**Comment by diego-plan9** [comment]
Reopening after #186 and #188 and moved to the proposal project - it seems we are getting some assertion failures when using `v2` and the endpoints with h/g/p (ie. the default ones to be used after the changes):

```
>       self.assertNotIn('backend', job_excluded)
E       AssertionError: 'backend' unexpectedly found in {...}
```

@jyu00 , can you take another look and check with the API team?

**Comment by diego-plan9** [comment]
Moved to `0.3.1` milestone - after #244, the tests are temporarily skipped.

**Comment by jyu00** [comment]
@diego-plan9 Do we even need these tests? And the `included_fields` and `excluded_fields` parameters for `AccountClient.job_get()`? I don't think they're used at all.

**Comment by diego-plan9** [comment]
Yes, let's remove the tests and the parameters (recursively).

**Comment by diego-plan9** [comment]
Fixed via #380
